modules/lambda-registrator/main.tf

```
@@ -126,9 +155,34 @@ resource "aws_iam_role_policy_attachment" "lambda_logs" {
   role       = aws_iam_role.registration.name
   policy_arn = aws_iam_policy.policy.arn
 }
+resource "random_id" "repo_id" {
+  byte_length = 4
+}
+
+resource "aws_ecr_repository" "lambda-registrator" {
+  count        = var.enable_auto_publish_ecr_image ? 1 : 0
+  name         = local.ecr_repo_name
+  force_delete = true
+}
+
+resource "docker_image" "lambda_registrator" {
+  count = var.enable_auto_publish_ecr_image ? 1 : 0
+  name  = var.consul_lambda_registrator_image
+}
+
+resource "docker_tag" "lambda_registrator_tag" {
+  count        = var.enable_auto_publish_ecr_image ? 1 : 0
+  source_image = docker_image.lambda_registrator[count.index].name
+  target_image = local.generated_ecr_image_uri
+}
+resource "docker_registry_image" "push_image" {
```

style nit: add a newline between blocks.

```suggestion

resource "docker_registry_image" "push_image" {
```